### PR TITLE
[Design] Show timeline behind expanded track groups

### PR DESCRIPTION
**Pull request by ALevansSamsung**
#### What it does

Show summary tracks when track groups are expanded rather then dead space

Based on the Argo designs it seems that rather than highlighting expanded track groups in this way, we would just show an empty track content.  From my perspective we could instead display the track group's summary track.  
It would look something like this: 
image

image

Also markers like notes would show through like normal tracks
image

This will fix this issue: https://github.com/android-graphics/sokatoa/issues/519
May also fix: https://github.com/android-graphics/sokatoa/issues/469

### CODE REVIEW

**Review by cdamus — CHANGES_REQUESTED** [review]
Looks as advertised: the grid is always continuous and the summary track (when present) always shows.

But the selection behaviour of groups is wonky. It requires multiple clicks to deselect and even then, sometimes it doesn't deselect all members of the group. Moreover, when initially drawing the selection, if it only overlaps some of the tracks of an expanded group the selection nonetheless takes in all of the other tracks in that group that I didn't draw the mouse over. See the very beginning of this screen grab for that selection rectangle behaviour.

CleanShot 2024-07-10 at 13 39 14

**Review by cdamus — COMMENTED** [review]
It's not clear to me what has changed ... the selection state of a group as controlled at the group level is still wildly inconsistent.

On the matter of the initial selection range, on reflection I think there's no issue there: the rubber-band is clearly previewing that the whole group will be selected when dragging the pointer over a part of it, so I think that's a clear communication to the user of what will happen.

CleanShot 2024-07-11 at 08 40 18

**Comment by ALevansSamsung** [comment]
> It's not clear to me what has changed ... the selection state of a group as controlled at the group level is still wildly inconsistent.

My latest change was suppose to make the checkboxes select and deselect children recursively.  I'll test again.

**Comment by ALevansSamsung** [comment]
Ok, it seems that scrolling over a track group and then also its children causes some tracks to be saved as selected multiple times therefore when they are deselected they are only deselecting the one iteration.

**Comment by cdamus** [comment]
> Ok, it seems that scrolling over a track group and then also its children causes some tracks to be saved as selected multiple times therefore when they are deselected they are only deselecting the one iteration.

Oh yeah, that makes sense. In fact, my GIF I think captured this? I notice that the tracks in the "Memory Usage" group that had sticky selection were the ones that my mouse pointer actually dragged over. Others that were included in the extended selection rubber-band were not so sticky.

File: ui/src/frontend/panel_container.ts [hunk 1]
```diff
@@ -158,20 +158,20 @@ export class PanelContainer implements m.ClassComponent<Attrs> {
         startY,
         endY);
     // Get the track ids from the panels.
-    const tracks = [];
+    const tracks = new Set<string>();
```
Comment (cdamus) [inline ui/src/frontend/panel_container.ts:161]: Nice! That's so simple 😀

**Review by cdamus — APPROVED** [review]
Yup, that did it. Thanks!

**Pull request merged**